openslides/core/static/js/core/projector.js

```diff
@@ -185,6 +188,45 @@ angular.module('OpenSlidesApp.core.projector', ['OpenSlidesApp.core'])
     }
 ])
 
+.controller('SlideSpeakerOverlayCtrl', [
+    '$scope',
+    'Motion',
+    'Assignment',
+    'Agenda',
+    'Topic',
+    'Projector',
+    'User',
+    function($scope, Motion, Assignment, Agenda, Topic, Projector, User) {
+        // Attention! Each object that is used here has to be dealt on server side.
+        // Add it to the coresponding get_requirements method of the ProjectorElement
+        // class.
+
+        $scope.visible = $scope.element.visible;
+        $scope.displayedElement = function() {
+            var displayedElement = [];
+            angular.forEach($scope.$parent.elements, function(element){
+                if (element.name == 'motions/motion') {
+                    displayedElement = ['motion', element.id];
+                } else if (element.name =='topics/topic') {
+                    displayedElement = ['topic', element.id];
+                } else if (element.name == 'assignments/assignment') {
+                    displayedElement = ['assignment', element.id];
+                }
+            });
+            if (displayedElement[0] == 'motion') {
```

Comment:
What about a switch case block? What happens if displayedElement is an empty list? Motion.get(undefined).agenda_item looks like an error ...